docs/content/preview/explore/change-data-capture/debezium-connector-yugabytedb.md

```diff
@@ -911,6 +911,26 @@ You can send this configuration with a `POST` command to a running Kafka Connect
 * Reads the transaction log.
 * Streams change event records to Kafka topics.
 
+{{< note title="Note" >}}
+
+We use a custom record extractor in order to make the sinks understand the format we are sending the data in i.e. `YBExtractNewRecordState`. For example, if you are using a JDBC Sink Connector, you need to add two more properties to the sink configuration:
+
+| Property | Value |
+| :--- | :--- |
+| `transforms` | `unwrap` |
+| `transforms.unwrap.type` | `io.debezium.connector.yugabytedb.transforms.YBExtractNewRecordState` |
+
+More about `YBExtractNewRecordState` is mentioned in the next section.
+{{< /note >}}
+
+#### `YBExtractNewRecordState` SMT
+
+Unlike the Debezium Connector for PostgreSQL, we only send the `after` image of the "set of columns" that are modified. PostgreSQL sends the complete `after` image of the row which has changed. So by default if the column was not changed, it is not a part of the payload we send and the default value is set to `null`.
+
+Now, to differentiate between the case where a column was set `null` or was not modified, we change the value type to a struct. In this structure, an unchanged column will be `{'value': null}`, whereas a changed column that was set to `null` will be `{'value': null, 'set': true}`.
+
+We have provided an SMT (Single Message Transformer for Kafka) `YBExtractNewRecordState` that interprets these values and sends the record in the format i.e removes the unmodified columns from the JSON message. In this way, the record which is transformed by 'YBExtractNewRecordState' is compatible with all the sink implementations. We took this approach so that the schema doesn't change with each new record and it can work with "Schema Registry". For example, if we keep changing the record to only include the value of columns then the schema of each record will be different (the total number unique schemas will be a result of making all possible combinations of columns) and thus we would require sending 'schema' with all the records. This will not work with "Schema Registry" where once a schema is registered, the record can only contain payload with schema 'version' and Kafka will make sure that it is interpreted correctly.
```

Comment:
I'm not sure I completely follow the logic, so please make sure my rewording here makes sense!
```suggestion
A schema registry requires that, once a schema is registered, records must contain only payloads with that schema version. If you're using a schema registry, the YugabyteDB Debezium connector's approach can be problematic, as the schema may change with every message. For example, if we keep changing the record to only include the value of modified columns, the schema of each record will be different (the total number unique schemas will be a result of making all possible combinations of columns) and thus would require sending a schema with every record.

To avoid this problem when you're using a schema registry, use the `YBExtractNewRecordState` SMT (Single Message Transformer for Kafka), which interprets these values and sends the record in the correct format (by removing the unmodified columns from the JSON message). Records transformed by `YBExtractNewRecordState` are compatible with all sink implementations. This approach ensures that the schema doesn't change with each new record and it can work with a schema registry.
```